client/main.lua

```diff
@@ -161,36 +177,105 @@ local function busGarage()
+
+                    CreateThread(function()
+                        repeat
+                            Wait(0)
+
+                            local inVeh = whitelistedVehicle()
+
+                            if not inVeh then
+                                lib.showTextUI(Lang:t('info.busstop_text'))
+
+                                if IsControlJustReleased(0, 38) then
+                                    busGarage()
+
+                                    lib.hideTextUI()
+                                    break
+                                end
+                            else
+                                lib.showTextUI(Lang:t('info.bus_stop_work'))
+
+                                if IsControlJustReleased(0, 38) then
+                                    if not NpcData.Active or NpcData.Active and not NpcData.NpcTaken then
+                                        if IsPedInAnyVehicle(cache.ped, false) then
+                                            BusData.Active = false
+
+                                            DeleteVehicle(cache.vehicle)
+                                            RemoveBlip(NpcData.NpcBlip)
+
+                                            lib.hideTextUI()
+
+                                            resetNpcTask()
+                                            break
+                                        end
+                                    else
+                                        QBCore.Functions.Notify(Lang:t('error.drop_off_passengers'), 'error')
+                                    end
+                                end
+                            end
+                        until not inRange
+                    end)
+                end,
+                onExit = function()
+                    lib.hideTextUI()
+
+                    inRange = false
+                end
+            })
+        end
+    else
+        if VehicleZone then
+            VehicleZone:remove()
+        end
+    end
+end
+
 RegisterNetEvent("qb-busjob:client:TakeVehicle", function(data)
     local coords = Config.Location
-    if (BusData.Active) then
+
+    if BusData.Active then
         QBCore.Functions.Notify(Lang:t('error.one_bus_active'), 'error')
         return
     else
         QBCore.Functions.TriggerCallback('QBCore:Server:SpawnVehicle', function(netId)
             local veh = NetToVeh(netId)
+
             SetVehicleNumberPlateText(veh, Lang:t('info.bus_plate') .. tostring(math.random(1000, 9999)))
-            exports['LegacyFuel']:SetFuel(veh, 100.0)
-            closeMenuFull()
+            SetVehicleFuelLevel(veh, 100.0)
+
+            lib.hideContext()
+
             TaskWarpPedIntoVehicle(cache.ped, veh, -1)
+
             TriggerEvent("vehiclekeys:client:SetOwner", QBCore.Functions.GetPlate(veh))
+
             SetVehicleEngineOn(veh, true, true, false)
         end, data.model, coords, true)
+
         Wait(1000)
+
         TriggerEvent('qb-busjob:client:DoBusNpc')
     end
```

Comment:
you could actually replace the else on line ```252``` with an end to save on indentation.